### ISSUE TITLE
Fix the issue with softmax classifier

### ISSUE THREAD

**Issue by thushv89** [thushv89/AutoEncorder_Simple#11]
Fixed the issue with the softmax classifier. The problem was, in the gradient descent I was doing this
delta[i] = y_vec[i]-actv[k] where y=k (y_vec is vectorized form of y, actv is current output activation vector  for y)

This is WRONG! Correct implementation is
delta[i] = y_vec[i] - actv[i]
